sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -674,6 +674,8 @@ nonReserved
     | AT | NULLS | OVERWRITE | ALL | ALTER | AS | BETWEEN | BY | CREATE | DELETE
     | DESCRIBE | DROP | EXISTS | FALSE | FOR | GROUP | IN | INSERT | INTO | IS |LIKE
     | NULL | ORDER | OUTER | TABLE | TRUE | WITH | RLIKE
+    | AND | CASE | CAST | CROSS | DISTINCT | DIV | ELSE | END | FROM | FUNCTION | HAVING | INTERVAL | JOIN|
```

Comment:
cc @hvanhovell 

Does adding things to this list cause any problems (performance?)?